snuba/querylog/query_metadata.py

```diff
@@ -15,6 +22,107 @@ class QueryStatus(Enum):
     TIMEOUT = "timeout"
 
 
+QUERY_ERROR_MAPPINGS = {
+    ErrorCodes.TOO_SLOW: QueryStatus.TIMEOUT,
+    ErrorCodes.TIMEOUT_EXCEEDED: QueryStatus.TIMEOUT,
+    ErrorCodes.SOCKET_TIMEOUT: QueryStatus.TIMEOUT,
+    ErrorCodes.NETWORK_ERROR: QueryStatus.TIMEOUT,
+}
+
+
+def get_query_status_from_error_codes(code: ErrorCodes) -> QueryStatus | None:
+    return QUERY_ERROR_MAPPINGS.get(code)
+
+
+class SLOStatus(Enum):
```

Comment:
I agree the names could be a bit better. Personally, I would refrain from using terms like `SLO` in code altogether. `SLO` is a service concern rather than a software concern. The service concern would be derived from a software concern.
```suggestion
class RequestStatus(Enum):
```
or
```suggestion
class APIStatus(Enum):
```